scipy/stats/_continuous_distns.py

```diff
@@ -678,6 +678,8 @@ def _pdf(self, x, a, b):
         #                     gamma(a+b) * x**(a-1) * (1-x)**(b-1)
         # beta.pdf(x, a, b) = ------------------------------------
         #                              gamma(a)*gamma(b)
+        message = "overflow encountered in _beta_pdf"
+        warnings.filterwarnings('ignore', message=message)
```

Comment:
If this is necessary, it should be handled with [`numpy.errstate`](https://numpy.org/doc/stable/reference/generated/numpy.errstate.html) instead of using the `warnings` module.

Reply:
Why?

Reply:
Tried it locally and it passes, I will commit.

Is `np.errstate` doing the same and just a convenience for consistency??